src/modules/add-remove-list.js

```diff
@@ -0,0 +1,74 @@
+// VARIABLES
+const listContainer = document.getElementById('list-container');
+let listArray = localStorage.getItem('list') ? JSON.parse(localStorage.getItem('list')) : [];
+let counter = 0;
+let counter2 = 0;
+
+// UPDATE
+const updateLocalStorage = () => {
+  localStorage.setItem('list', JSON.stringify(listArray));
+};
+
+const updateIndexes = () => {
+  listArray.forEach((element) => {
+    element.indexVal = counter2;
+    counter2 += 1;
+  });
+  counter2 = 0;
+  updateLocalStorage();
+};
+
+export const updateDescription = (index, desc) => {
+  listArray[index].description = desc;
+  updateLocalStorage();
+};
+
+// LOAD
+export const renderList = () => {
+  listArray.forEach((element) => {
+    listArray[counter].indexVal = counter;
+    element.indexVal = counter;
+    listContainer.insertAdjacentHTML('beforeend', `
+    <li id="${element.indexVal}">
+      <input class="checkbox" type="checkbox">
+      <textarea class="desc" maxlength="255" data-ta="${element.indexVal}">${element.description}</textarea>
+      <i class="move fa-solid fa-ellipsis-vertical"></i>
+      <i class="erase fa-solid fa-trash" data-id="${element.indexVal}"></i>   
+    </li>
+  `);
+    counter += 1;
+  });
+};
+
+// REMOVE
+export const removeItem = (id) => {
+  const selectedElement = document.getElementById(`${id}`);
+  listContainer.removeChild(selectedElement);
+  listArray = listArray.filter((item) => item.indexVal !== +id);
+  updateIndexes();
+};
+
+// ADD
+const addNew = () => {
+  const newItem = listArray[listArray.length - 1];
+  listContainer.insertAdjacentHTML('beforeend', `
+  <li id="${newItem.indexVal}">
+    <input class="checkbox" type="checkbox">
+    <textarea class="desc" maxlength="255" data-ta="${newItem.indexVal}">${newItem.description}</textarea>
+    <i class="move fa-solid fa-ellipsis-vertical"></i>
+    <i class="erase fa-solid fa-trash" data-id="${newItem.indexVal}"></i>
+  </li>
+  `);
```

Comment:
- [ ] Great job so far 👏🏻 However the property index should be set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5). Actually, your first book has indexVal equal to 0; instead of 1. Kindly address this 💯 👍🏻 
- Make sure that, deleting a task update all remaining items' indexes, so they represent the current list order and are unique. 👍🏻